src/lib/y2network/sysconfig_routes_file.rb

```diff
@@ -0,0 +1,194 @@
+    # @param netmask_str [String] Netmask; {MISSING_VALUE} means than no netmask was specified
+    # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+    def build_ip(ip_str, netmask_str = MISSING_VALUE)
+      return nil if ip_str == MISSING_VALUE
+
+      ip = IPAddr.new(ip_str)
+      netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+    end
+
+    # Build a route given a hash from the SCR agent
+    #
+    # @param hash [Hash] Hash from the `.routes` SCR agent
+    # @return Route
+    def build_route(hash)
+      # TODO: check whether the iface is configured in the system
+      iface = Interface.new(hash["device"])
+      # normalized SCR output contains either subnet mask or /<prefix length> under
+      # "netmask" key
+      # TODO: this should be improved in normalize_routes
+      mask = hash["netmask"] =~ /\/[0-9]+/ ? hash["netmask"][1..-1] : hash["netmask"]
+
+      Y2Network::Route.new(
+        to:        hash["destination"] != DEFAULT_DEST ? build_ip(hash["destination"], mask) : :default,
+        interface: iface,
+        gateway:   build_ip(hash["gateway"], MISSING_VALUE),
+        options:   hash["extrapara"] || ""
+      )
+    end
+
+    # SCR agent for routes files definition
+    def ifroute_term(path)
+      raise ArgumentError if path.nil? || path.empty?
+
+      non_empty_str_term = Yast.term(:String, "^ \t\n")
+      whitespace_term = Yast.term(:Whitespace)
+      optional_whitespace_term = Yast.term(:Optional, whitespace_term)
+      routes_content_term = Yast.term(
+        :List,
+        Yast.term(
+          :Tuple,
+          Yast.term(
+            :destination,
+            non_empty_str_term
+          ),
+          whitespace_term,
+          Yast.term(:gateway, non_empty_str_term),
+          whitespace_term,
+          Yast.term(:netmask, non_empty_str_term),
+          optional_whitespace_term,
+          Yast.term(
+            :Optional,
+            Yast.term(:device, non_empty_str_term)
+          ),
+          optional_whitespace_term,
+          Yast.term(
+            :Optional,
+            Yast.term(
+              :extrapara,
+              Yast.term(:String, "^\n")
+            )
+          )
+        ),
+        "\n"
+      )
+
+      Yast.term(
+        :ag_anyagent,
+        Yast.term(
+          :Description,
+          Yast.term(:File, path),
+          "#\n",
+          false,
+          routes_content_term
+        )
+      )
+    end
+
+    # Registers SCR agent which is used for accessing particular ifroute-device
+    # file
+    #
+    # @param device [String] full path to a file in routes format (e.g. /etc/sysconfig/network/ifroute-eth0)
+    # @return [Path] SCR path of the agent
+    # @raise  [RuntimeError] if it fails
+    def register_ifroute_agent_for_path(path)
+      # /etc/sysconfig/network/ifroute-eth0 define .ifroute-eth0 agent
+      # TODO: collisions not handled
```

Comment:
well, you can also unregister scr path. So in reality you can use some temporary scr path just for loading and then simply unregister it again. See real world example at https://github.com/yast/yast-bootloader/blob/78f31a9f30e42641c1147d199cac103aa1a4380f/src/lib/bootloader/sysconfig.rb#L144

Reply:
still true?